License filename matching: check for word boundaries around the word 'license' or 'licence'

jayaddison opened the pull request:
Resolves #126.

jayaddison committed:
Add illustrative test case

jayaddison committed:
Check for word boundaries around the word 'license' or 'licence' when looking for potential license files in project directories

xz64 commented on src/LicenseTextReader.ts:
```diff
@@ -85,7 +85,10 @@ class LicenseTextReader {
+      if (
+        /\blicen[cs]e\b/i.test(path) &&
+        !this.fileSystem.isDirectory(filePath)
+      ) {
```
I think it would be safer to keep the original regex first, and if nothing is found from the first regex, then use the other regex.

jayaddison replied:
That makes sense, thanks - change applied (including a misstep where I considered moving the `isDirectory` test; that could be relatively expensive on some systems I suppose).

jayaddison committed:
Revert "Check for word boundaries around the word 'license' or 'licence' when looking for potential license files in project directories"

This reverts commit b673731efe0adfa7b06c4e829d6c391fa9714cbf.

jayaddison committed:
Refactor: separate pattern-check and is-directory-check into separate conditional clauses

jayaddison committed:
Check for word boundaries around the word 'license' or 'licence' when looking for potential license files in project directories

This is an alternative implementation of commit b673731efe0adfa7b06c4e829d6c391fa9714cbf.

jayaddison committed:
Revert conditional ordering: perform pattern-checks before is-directory-check

Interleaves implementation approaches from commit b673731efe0adfa7b06c4e829d6c391fa9714cbf and commit 1ea317b318a1a3fe74a79294e464594c8efb4def.

jayaddison commented:
Ready for re-review when you have time, @xz64

jayaddison commented:
I think that it would have been more appropriate for me to open this pull request from a fork of this repository that belongs to the [`openculinary`](https://github.com/openculinary/) organization on GitHub (since it affects code developed in my capacity as an employee of it), so I'm going to close this and re-open it from there soon.
